oasislmf/pytools/common/data.py

```diff
@@ -141,3 +156,183 @@ def load_as_array(dir_path, name, _dtype, must_exist=True):
+    Args:
+        run_dir (str | os.PathLike): Path to input files dir
+        filename (str | os.PathLike): occurrence binary file name
+    Returns:
+        occ_map (ndarray[occ_map_dtype]): numpy map of event_id, period_no, occ_date_id from the occurrence file
+    """
+    from .event_stream import mv_read
+    try:
+        occurrence_fp = Path(run_dir, filename)
+        fin = np.memmap(occurrence_fp, mode="r", dtype="u1")
+        cursor = 0
+        valid_buff = len(fin)
+
+        if valid_buff - cursor < np.dtype(np.int32).itemsize:
+            raise RuntimeError("Error: broken occurrence file, not enough data")
+        date_opts, cursor = mv_read(fin, cursor, np.int32, np.dtype(np.int32).itemsize)
+
+        date_algorithm = date_opts & 1
+        granular_date = date_opts >> 1
+
+        # (event_id: int, period_no: int, occ_date_id: int)
+        record_size = np.dtype(np.int32).itemsize * 3
+        # (event_id: int, period_no: int, occ_date_id: long long)
+        if granular_date:
+            record_size = np.dtype(np.int32).itemsize * 2 + np.dtype(np.int64).itemsize
+        # Should not get here
+        if not date_algorithm and granular_date:
+            raise RuntimeError("FATAL: Unknown date algorithm")
+
+        # Extract no_of_periods
+        if valid_buff - cursor < np.dtype(np.int32).itemsize:
+            raise RuntimeError("Error: broken occurrence file, not enough data")
+        no_of_periods, cursor = mv_read(fin, cursor, np.int32, np.dtype(np.int32).itemsize)
+
+        num_records = (valid_buff - cursor) // record_size
+        if (valid_buff - cursor) % record_size != 0:
+            logger.warning(
+                f"Occurrence File size (num_records: {num_records}) does not align with expected record size (record_size: {record_size})"
+            )
+
+        occ_map_dtype = np.dtype([
+            ("event_id", np.int32),
+            ("period_no", np.int32),
+            ("occ_date_id", np.int32),
+        ])
+        if granular_date:
+            occ_map_dtype = np.dtype([
+                ("event_id", np.int32),
+                ("period_no", np.int32),
+                ("occ_date_id", np.int64),
+            ])
+
+        occ_map = np.zeros(num_records, dtype=occ_map_dtype)
+
+        for i in range(num_records):
+            if valid_buff - cursor < record_size:
+                break  # Not enough data left
+            event_id, cursor = mv_read(fin, cursor, np.int32, np.dtype(np.int32).itemsize)
+            period_no, cursor = mv_read(fin, cursor, np.int32, np.dtype(np.int32).itemsize)
+            if granular_date:
+                occ_date_id, cursor = mv_read(fin, cursor, np.int64, np.dtype(np.int64).itemsize)
+            else:
+                occ_date_id, cursor = mv_read(fin, cursor, np.int32, np.dtype(np.int32).itemsize)
+            occ_map[i] = (event_id, period_no, occ_date_id)
+
+        return occ_map, date_algorithm, granular_date, no_of_periods
+    except Exception as e:
```

Comment:
Just let the error be raised, this exception handling doesn't add any information